tools/src/main/java/org/apache/kafka/tools/CompatibilityTest.java

```diff
@@ -0,0 +1,359 @@
+        Future<RecordMetadata> future2 = producer.send(record2, null);
+        producer.flush();
+        future1.get();
+        future2.get();
+        producer.close();
+    }
+
+    private static class OffsetsForTime {
+        Map<TopicPartition, OffsetAndTimestamp> result;
+
+        @Override
+        public String toString() {
+            return Utils.mkString(result);
+        }
+    }
+
+    public static class CompatibilityTestDeserializer implements Deserializer<byte[]>, ClusterResourceListener {
+        private final boolean expectClusterId;
+
+        CompatibilityTestDeserializer(boolean expectClusterId) {
+            this.expectClusterId = expectClusterId;
+        }
+
+        @Override
+        public void configure(Map<String, ?> configs, boolean isKey) {
+            // nothing to do
+        }
+
+        @Override
+        public byte[] deserialize(String topic, byte[] data) {
+            return data;
+        }
+
+        @Override
+        public void close() {
+            // nothing to do
+        }
+
+        @Override
+        public void onUpdate(ClusterResource clusterResource) {
+            if (expectClusterId) {
+                if (clusterResource.clusterId() == null) {
+                    throw new RuntimeException("expected cluster id to be " +
+                        "supported, but it was null.");
+                }
+            } else {
+                if (clusterResource.clusterId() != null) {
+                    throw new RuntimeException("expected cluster id to be " +
+                        "null, but it was supported.");
+                }
+            }
+        }
+    }
+
+    public void testConsume(final long prodTimeMs) throws Exception {
+        Properties consumerProps = new Properties();
+        consumerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, testConfig.bootstrapServer);
+        CompatibilityTestDeserializer deserializer = new CompatibilityTestDeserializer(testConfig.expectClusterId);
+        final KafkaConsumer<byte[], byte[]> consumer =
+                new KafkaConsumer<byte[], byte[]>(consumerProps, deserializer, deserializer);
+        final List<PartitionInfo> partitionInfos = consumer.partitionsFor(testConfig.topic);
+        if (partitionInfos.size() < 1)
+            throw new RuntimeException("expected at least one partition for topic " + testConfig.topic);
+        final Map<TopicPartition, Long> timestampsToSearch = new HashMap<>();
+        final LinkedList<TopicPartition> topicPartitions = new LinkedList<>();
+        for (PartitionInfo partitionInfo: partitionInfos) {
+            TopicPartition topicPartition = new TopicPartition(partitionInfo.topic(), partitionInfo.partition());
+            timestampsToSearch.put(topicPartition, prodTimeMs);
+            topicPartitions.add(topicPartition);
+        }
+        final OffsetsForTime offsetsForTime = new OffsetsForTime();
```

Comment:
Yes, I know we assign `result` in the block, but I don't understand what the aim is as we don't do anything else with it. It seems like we can remove the assignment and the behaviour remains the same. What am I missing?